Tests/TuistKitTests/Cache/CacheControllerTests.swift

```diff
@@ -122,4 +122,63 @@ final class CacheControllerTests: TuistUnitTestCase {
         XCTAssertEqual(artifactBuilder.invokedBuildWorkspacePathParametersList[1].target, bTarget)
         XCTAssertEqual(artifactBuilder.invokedBuildWorkspacePathParametersList[2].target, cTarget)
     }
+
+    func test_filtered_cache_builds_and_caches_the_frameworks() throws {
+        // Given
+        let path = try temporaryPath()
+        let xcworkspacePath = path.appending(component: "Project.xcworkspace")
+        let project = Project.test(path: path, name: "Cache")
+        let targetNames = ["foo", "bar", "baz"].shuffled()
+        let aTarget = Target.test(name: targetNames[0])
+        let bTarget = Target.test(name: targetNames[1])
+        let cTarget = Target.test(name: targetNames[2])
+        let aFrameworkPath = path.appending(component: "\(aTarget.name).framework")
+        let bFrameworkPath = path.appending(component: "\(bTarget.name).framework")
+        let cFrameworkPath = path.appending(component: "\(cTarget.name).framework")
+        try FileHandler.shared.createFolder(aFrameworkPath)
+        try FileHandler.shared.createFolder(bFrameworkPath)
+        try FileHandler.shared.createFolder(cFrameworkPath)
+
+        let aTargetNode = TargetNode.test(project: project, target: aTarget)
+        let bTargetNode = TargetNode.test(project: project, target: bTarget, dependencies: [aTargetNode])
+        let cTargetNode = TargetNode.test(project: project, target: cTarget, dependencies: [bTargetNode])
+        let nodeWithHashes = [
+            aTargetNode: "\(aTarget.name)_HASH",
+            bTargetNode: "\(bTarget.name)_HASH",
+            cTargetNode: "\(cTarget.name)_HASH",
+        ]
+        let graph = Graph.test(projects: [project],
+                               targets: nodeWithHashes.keys.reduce(into: [project.path: [TargetNode]()]) { $0[project.path]?.append($1) })
+
+        manifestLoader.manifestsAtStub = { (loadPath: AbsolutePath) -> Set<Manifest> in
+            XCTAssertEqual(loadPath, path)
+            return Set(arrayLiteral: .project)
+        }
+        generator.generateWithGraphStub = { (loadPath, _) -> (AbsolutePath, Graph) in
+            XCTAssertEqual(loadPath, path)
+            return (xcworkspacePath, graph)
+        }
+        generator.generateStub = { (loadPath, _) -> AbsolutePath in
+            XCTAssertEqual(loadPath, path)
+            return xcworkspacePath
+        }
+        graphContentHasher.stubbedContentHashesResult = nodeWithHashes
+        artifactBuilder.stubbedCacheOutputType = .xcframework
+
+        try subject.cache(path: path, targetsToFilter: [bTarget.name])
```

Comment:
Nitpick: I'd add `// When` comment before this line

Reply:
I followed the same structure as the test above, would you change both?

Reply:
Yes, I think it was just forgotten there 🙂

Reply:
Ah sorry, I read "Then" not "When" hehe